src/chemistry/utils/aircraft_emit.F90

```diff
@@ -1,458 +1,632 @@
+           aircraft_h2o_fldname, aircraft_h2o_datafile, aircraft_h2o_meshfile, &
+           aircraft_h2o_year_first, aircraft_h2o_year_last, aircraft_h2o_year_align, &
+           aircraft_slant_dist_fldname, aircraft_slant_dist_datafile, aircraft_slant_dist_meshfile, &
+           aircraft_slant_dist_year_first, aircraft_slant_dist_year_last, aircraft_slant_dist_year_align
+      !-----------------------------------------------------------------------------
+
+      ! Read namelist
+      if (masterproc) then
+         open( newunit=unitn, file=trim(nlfile), status='old' )
+         call find_group_name(unitn, 'aircraft_emit_nl', status=ierr)
+         if (ierr == 0) then
+            read(unitn, aircraft_emit_nl, iostat=ierr)
+            if (ierr /= 0) then
+               call endrun(subname // ':: ERROR reading namelist')
+            end if
          end if
+         close(unitn)
+
+         forcing(1)%fldname    = aircraft_co2_fldname
+         forcing(1)%datafile   = aircraft_co2_datafile
+         forcing(1)%meshfile   = aircraft_co2_meshfile
+         forcing(1)%year_first = aircraft_co2_year_first
+         forcing(1)%year_last  = aircraft_co2_year_last
+         forcing(1)%year_align = aircraft_co2_year_align
+
+         forcing(2)%fldname    = aircraft_h2o_fldname
+         forcing(2)%datafile   = aircraft_h2o_datafile
+         forcing(2)%meshfile   = aircraft_h2o_meshfile
+         forcing(2)%year_first = aircraft_h2o_year_first
+         forcing(2)%year_last  = aircraft_h2o_year_last
+         forcing(2)%year_align = aircraft_h2o_year_align
+
+         forcing(3)%fldname    = aircraft_slant_dist_fldname
+         forcing(3)%datafile   = aircraft_slant_dist_datafile
+         forcing(3)%meshfile   = aircraft_slant_dist_meshfile
+         forcing(3)%year_first = aircraft_slant_dist_year_first
+         forcing(3)%year_last  = aircraft_slant_dist_year_last
+         forcing(3)%year_align = aircraft_slant_dist_year_align
       end if
-      close(unitn)
-      call freeunit(unitn)
-   end if
 
-#ifdef SPMD
-   ! Broadcast namelist variables
-   call mpibcast(aircraft_specifier,len(aircraft_specifier(1))*N_AERO,     mpichar, 0, mpicom)
-   call mpibcast(aircraft_datapath, len(aircraft_datapath),                mpichar, 0, mpicom)
-   call mpibcast(aircraft_type,     len(aircraft_type),                    mpichar, 0, mpicom)
-#endif
+      n_aero_loop: do nf = 1,N_AERO
+         ! Broadcast namelist variables
+         call mpi_bcast(forcing(nf)%fldname,len(forcing(nf)%fldname), mpi_character, masterprocid, mpicom, ierr)
+         if (ierr /= 0) call endrun(subname//": FATAL: mpi_bcast: forcing(nf)%fldname")
+
+         if (forcing(nf)%fldname /= ' ') then
+            call mpi_bcast(forcing(nf)%datafile, len(forcing(nf)%datafile), mpi_character, masterprocid, mpicom, ierr)
+            if (ierr /= 0) call endrun(subname//": FATAL: mpi_bcast: forcing(nf)%datapath")
```

Comment:
These messages don't tell you which field-type failed. Could you use this instead (subst for `nf`)?
```suggestion
            if (ierr /= 0) call endrun(subname//": FATAL: mpi_bcast: forcing("//int2str(nf)//")%datapath")
```
You will need `use string_utils, only: int2str``.

Reply:
Good suggestion. Done.